<launch>
<!-- Launch file for Mapping -->

	<arg name="rviz" default="true" />

    <rosparam file="$(find backend_optimization)/config/test_mid360.yaml" command="load" />

	<param name="showOptimizedPose" type="bool" value="true" />
	<param name="globalMapVisualizationSearchRadius" type="double" value="1000" />
	<param name="globalMapVisualizationPoseDensity" type="double" value="3" />
	<param name="globalMapVisualizationLeafSize" type="double" value="0.2" />

    <node pkg="backend_optimization" type="pgo" name="system" output="screen" /> 

	<group if="$(arg rviz)">
		<node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find backend_optimization)/rviz_cfg/pgo.rviz" />
	</group>

</launch>